pkg/resource/provider/host.go

```diff
@@ -34,6 +37,9 @@ type HostClient struct {
 
 // NewHostClient dials the target address, connects over gRPC, and returns a client interface.
 func NewHostClient(addr string) (*HostClient, error) {
+	// GRPC info logging to stdout introduces a race condition with the printing of the port, so drop them.
+	// See https://github.com/pulumi/pulumi/issues/7156
+	grpclog.SetLoggerV2(grpclog.NewLoggerV2(ioutil.Discard, os.Stderr, os.Stderr))
```

Comment:
It should not be the case that grpclog logs to stdout by default.  Per https://github.com/grpc/grpc-go/blob/v1.40.0/grpclog/loggerv2.go#L123-L146, logs are only to stderr by default, and only on errors by default.

If you are seeing something different than that, I suspect it must be due to using some different version of that module of some environmental configuration?

That said, since we are particularly incompatible with sending info logs to stdout, it does potentially make sense to try to override this - but perhaps we should set it back to the same thing grpclog does by default (`Discard`, `Discard`, `Stderr`)?

Reply:
updated